src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp

```diff
@@ -79,15 +80,7 @@ void ShenandoahScanRememberedTask::do_work(uint worker_id) {
   // set up thread local closure for shen ref processor
   _rp->set_mark_closure(worker_id, &cl);
   struct ShenandoahRegionChunk assignment;
-  bool has_work = _work_list->next(&assignment);
-  while (has_work) {
-#ifdef ENABLE_REMEMBERED_SET_CANCELLATION
-    // This check is currently disabled to avoid crashes that occur
-    // when we try to cancel remembered set scanning
-    if (heap->check_cancelled_gc_and_yield(_is_concurrent)) {
-      return;
-    }
-#endif
+  while (_work_list->next(&assignment)) {
```

Comment:
When a work item is picked up, it should be completed since it can't currently be placed back on the work list. Hence the code change, moving the check to the end of the loop. My expectation had been that this would fix the issue with prompt cancellation causing crashes previously. However, testing revealed that this was still causing crashes, so I left the cancellation commented out with the intention of following up on this in the fullness of time. 

I'll leave a comment to that effect as you suggest, and mark it with a TODO so it's easily flagged/found.

Reply:
Fixed.